Server.Logic/Email/EmailManager.cs

```diff
@@ -0,0 +1,61 @@
+using System.ComponentModel.DataAnnotations;
+using Server.Database;
+using Server.Logic.Abstract.Email;
+using MimeKit;
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Configuration;
+
+namespace Server.Logic.Email;
+
+internal class EmailManager  : IEmailManager{    
+    private readonly GitfoodContext _dbInfo;
+    private readonly IConfiguration _configuration;
+    private readonly String _email;
+    private readonly String _password;
+
+    public EmailManager(GitfoodContext dbInfo, IConfiguration configuration)
+    {
+        _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        _dbInfo = dbInfo ?? throw new ArgumentNullException(nameof(dbInfo));
+        _email = _configuration.GetSection("EmailConfig").GetSection("Email").Value;
+        _password = _configuration.GetSection("EmailConfig").GetSection("Password").Value;
+    }
+
+    public bool isBanned(string email)
+    {
+        var user = _dbInfo.Users.FirstOrDefault(e => e.Email == email);
+        return user != null && user.IsBanned;
+    }
+
+    public bool isValid(string email)
+    {
+        var emailValidator = new EmailAddressAttribute();
+        return emailValidator.IsValid(email);
+    }
+
+    public bool isVerified(string email)
+    {
+        var user = _dbInfo.Users.FirstOrDefault(e => e.Email == email);
+        return user != null && user.Verification == null;
+    }
+
+    public void sendVerificationEmail(string email, string verificationToken, string user)
+    {
+        var message = new MimeMessage();
+        message.From.Add(new MailboxAddress(name: "gitfood.fun", address: _email));
```

Comment:
Nie lepiej by było dać to do stałych albo w config (appsettingsy) name, port, subject oraz text, szczególnie że chyba cały interface storny mamy po angielsku o ile dobrze kojarze?

Reply:
I tak bierzemy w konstruktorze IConfiguration wiec raczej nie będzie to dużo zmian, jakbyśmy się wsm chcieli bawi to można dodać resourcey do wielu jezyków ale narazie chyba nie ma sensu się w to bawić